tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1023,7 +1029,11 @@ public:
       if (!dtorOnly)
          free(vec);
    }
-   void CommitCluster() final { fNWritten = 0; }
+   void CommitCluster() final {
+        Detail::RColumnElement<ClusterSize_t, EColumnType::kIndex> elemIndex(&fNWritten);
+        fColumns[0]->Append(elemIndex);
+        printf("just appended: %d\n", fNWritten);
+   }
```

Comment:
fNWritten is no longer reset here. I changed this to fix empty vectors in `CommitCluster`, but it might have had side effects elsewhere.

Reply:
`fNWritten` needs to be reset so that it starts with zero for the next cluster.  I would have thought that we can safely reset it at the end of `CommitCluster()` (after flushing it out to the offset column).  In which way would it break empty empty vectors?

Reply:
@jblomer, this came up when I got a debug line like: 
`idxStart: 3 | idxStop: 0 | size: -3` 
so I mistakenly thought we shouldn't reset `fNWritten` to make `idxStop = 3`. 

However, I think the problem is that `Append` calls in `CommitCluster` change index positions in `GetCollectionInfo`. Here is a small example of what I think is happening: 
```cpp 
auto model = RNTupleModel::Create(); 
auto floats = model->MakeField<std::vector<float>>("flts"); 
// vector of length 1
floats->push_back({1.0}); 
{
   RNTupleWriter ntuple(std::move(model),
      std::make_unique<RPageSinkFile>("myNTuple", fileGuard.GetPath(), RNTupleWriteOptions()));
   ntuple.Fill();
   ntuple.CommitCluster();
   ntuple.Fill();
}
```
so at the end we have two clusters. 

The existing (not using the changes in this PR) offset column is: 
`[1 1]` and `GetCollectionInfo` knows where cluster boundaries are and can set `idxStart` to 0 if need be. 

The new offset column (with resetting) is: 
`[0 1 0 1]` which I think breaks global indexing. 

Global indexing looks up `offset[index], offset[index + 1]`
which is OK for `globalIndex: 0 => [0, 1)` but when `globalIndex = 1` it no longer lines up properly with the offset column: `globalIndex: 1 => [1, 0)`. 

I think to get correct behaviour, we could try: 
- inserting a 0 on construction of the `RFieldVector` (ran into some trouble where I wasn't able to insert column values right away)
- append the written length value in `AppendImpl`
- reset (but not append) `fNWritten` in `CommitCluster` 

(as a side note, I am having trouble seeing the offset column directly, and would like to try and implement some sort of `Dump` function for columns :) )

Reply:
That is a very good point!  Good catch.

I think that treating the first cluster different from the others would not work either.  Let me think about it.  Maybe, it means that the way offsets currently work are better.

Reply:
Ok, thank you :) I think the current behaviour has some advantages and it's nice to hide all the tricky stuff inside `GetCollectionInfo`.